spec/wcc/contentful/model_builder_spec.rb

```diff
@@ -246,6 +246,23 @@
     expect(main_menu.hamburger.items[0].link.title).to eq('About')
   end
 
+  it 'stores backreference on linked type context' do
+    @schema = subject.build_models
+
+    # act
+    main_menu = WCC::Contentful::Model::Menu.find('FNlqULSV0sOy4IoGmyWOW')
+
+    # assert
+    hamburger = main_menu.hamburger
+    expect(hamburger.sys.context.backlinks).to eq([main_menu])
+
+    button = hamburger.items[0]
+    expect(button.sys.context.backlinks).to eq([hamburger, main_menu])
+
+    page = button.link
+    expect(page.sys.context.backlinks).to eq([button, hamburger, main_menu])
+  end
```

Comment:
@jpowell check this out!  Let's just expose this to the app in a more raw fashion, and the app can figure out what it means that the first backlink from the page is a button.

Reply:
I guess this is used in the `already_resolved` above?

Reply:
It's used to check for circular references.  I already had been storing an array like this for that purpose, I just exposed it on the context where it could potentially be useful.

I wrote that because I had initially been thinking about using the backlinks to avoid query hits inside the CanCan ability checks.